### DISCUSSION TITLE
Scheduler not starting when SQL not avaliable

### DISCUSSION THREAD

**Discussion by Roneri75** [quartznet/quartznet#1697] [Q&A]
So we have an .NET Core Web app running .NET 6 with the Scheduler in the ASP.net backend.
Everything is run on Windows Server in IIS.
When we startup the app the scheduler connects to the MSSQL server and everything is good, runs like a charm.
Normal setup is the App is on one server and the MSSQL server is on another server.

But, if the SQL server is not available for any reason when the App is starting and the Scheduler should connect we get this:
"Failure occured during job recovery. Failed to obtain DB connection from data source 'default' "
and that would be expected.
But the problem here is that then the scheduler id "dead" and we have no ide that it failed the startup, but the app is running and we have to manually go and do a Recycle on the AppPool to "restart" the scheduler, when the SQL is avaliable, and then it works.

What is the best way to handle this?
We need the scheduler to be up at all times.
But if the servers are restarted, and the SQL is not available when the Scheduler connects, the Scheduler will not work, or is any other case where the Scheduler can not connect to the SQL server.
I would like to stay way from like sending an email to someone if it does not work or alike.
The best way, for us, would be is we could "restart" the scheduler indefinitely, if this happens, until it connects or someone shuts it down on the server.

Any ides?

**Comment by lahma** [comment]
Here's some relevant discussion from before: https://github.com/quartznet/quartznet/issues/520
